### PR TITLE
Include String Matching Methods

### PR DESCRIPTION
**Pull request by tyler-spangler6**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary
This PR adds functions to implement a new string matching algorithm to use in a forthcoming revision of the AutosuggestField component.  This is not being used yet and will be released behind a feature flag when a revised addDisabilities page and AutosuggestField component is implemented.  These functions were tested in a codepen prototype during Veteran research with the goal of improving suggestions for Veterans as they enter conditions for disability compensation.  

## Related issue(s)
[#453](https://app.zenhub.com/workspaces/contention-classification-640a24db5dcd08457cebbc79/issues/gh/department-of-veterans-affairs/vagov-claim-classification/453)
- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#0000
- _Link to previous change of the code/bug (if applicable)_
department-of-veterans-affairs/vets-website#0000
- _Link to epic if not included in ticket_
department-of-veterans-affairs/va.gov-team#0000

## Testing done
- Unit tests are included for main search function and all accessory functions
- Test by running `yarn test:unit --log-level debug vets-website/src/platform/forms-system/test/js/utilities/ccStringSearch.unit.spec.js`

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Comment by lukey-luke** [comment]
For reference, the autosuggest component that we're extending/updating is [HERE](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/forms-system/src/js/fields/AutosuggestField.jsx#L40).

**Review by lukey-luke — COMMENTED** [review]
Love to see the test coverage with different length strings.

Only nitpick I have would be the file naming. I think naming it something to the tune of `addDisabilitiesXYZ` e.g. `addDisabilitiesStringSearch` might make more sense to other developers and makes it more obvious where this is used on the site

File: src/platform/forms-system/test/js/utilities/ccStringSearch.unit.spec.js [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    const expectedString = 'knee hurts,';
+    expect(stripCommonWords(testString)).to.eql(expectedString);
+  });
+});
+
+describe('lcsScoreByWordSum', () => {
+  it('should return integer score for similarity', () => {
+    const inputString = 'right acl tear in knee';
+    const disabilityWord = 'ACL tear (anterior cruciate ligament tear), right';
+    const expectedScore = 8;
```
Comment (lukey-luke) [inline src/platform/forms-system/test/js/utilities/ccStringSearch.unit.spec.js:37]: I think defining score in `lcsScoreByWordSum()` might help for updating the thresholds / tweaking scores in the future

**Review by lukey-luke — DISMISSED** [review]

File: yarn.lock [hunk 2]
```diff
@@ -15188,6 +15188,11 @@ node-gyp@^8.2.0:
     tar "^6.1.2"
     which "^2.0.2"
 
+node-lcs@^1.0.4:
```
Comment (pjhill) [inline yarn.lock:15191]: node-lcs has [no dependencies](https://www.npmjs.com/package/node-lcs?activeTab=dependencies), so it should be a pretty lightweight include.

**Review by pjhill — APPROVED** [review]

**Review by lukey-luke — APPROVED** [review]

**Review by micahchiang — APPROVED** [review]

**Pull request merged**